### PR TITLE
feat: Add dropout regularization, configurable learning rate, input validation and `model_summary()` to `model.py` [ GSSOC'26 ]

### LINKED ISSUE #1066: Feat: Add dropout regularization, configurable learning rate, input validation and `model_summary()` to `model.py`
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the feature

The `build_model()` function in `backend/ml/model.py` currently has no dropout, uses the string `"adam"` optimizer with no learning rate control, has no input validation, and no way to inspect the architecture without training. This makes the model prone to overfitting and hard to tune.

This feature adds:
- `dropout` parameter (default `0.2`) — applied after encoder and decoder LSTM layers via `Dropout()` to prevent the model memorizing training data, which would make the anomaly threshold meaningless on real sessions
- `learning_rate` parameter (default `0.001`) — replaces the `"adam"` string with an explicit `Adam(learning_rate=...)` so `train.py` can tune it
- Input guards with clear `ValueError` messages for invalid `units`, `dropout`, and `learning_rate` values
- Named the model `lstm_autoencoder` for easier identification in logs and summaries
- Added `model_summary()` helper — prints default architecture without needing to train
- Added module-level docstring

### Acceptance criteria (optional)

- [x] `build_model()` accepts `units`, `dropout`, and `learning_rate` parameters
- [x] `Dropout` layers appear after encoder and decoder in `model.summary()`
- [x] `build_model(dropout=1.5)` raises `ValueError`
- [x] `build_model(units=0)` raises `ValueError`
- [x] `build_model(learning_rate=-1)` raises `ValueError`
- [x] `model_summary()` prints the architecture without error
- [x] `python model.py` runs and prints the summary
- [x] `SEQ_LEN` and `N_FEATURES` constants remain unchanged

### Screenshots or mockups (if applicable)

_No response_

### Record

- [x] I agree to follow this project's Code of Conduct
- [x] I want to work on this issue

### PR DESCRIPTION
**Pull request by rudra3007-pro**
## What this PR does

Hardens `build_model()` in `backend/ml/model.py` by adding dropout regularization, a configurable learning rate, input validation, and a `model_summary()` helper.

Previously the model had no dropout — meaning it could memorize training data and produce a meaningless anomaly threshold at inference. The `"adam"` string optimizer also gave no control over learning rate for tuning in `train.py`.

**Changes in `backend/ml/model.py`:**
- Added `dropout` parameter (default `0.2`) — `Dropout()` layers inserted after encoder and decoder LSTMs
- Added `learning_rate` parameter (default `0.001`) — replaces `"adam"` string with explicit `Adam(learning_rate=...)`
- Added input guards — `ValueError` raised for invalid `units`, `dropout`, and `learning_rate` values
- Named the model `lstm_autoencoder` for clearer logs and summaries
- Added `model_summary()` — prints architecture without training
- Added module-level docstring
- `python model.py` now prints the model summary via `if __name__ == "__main__"`

## Type of change

- [x] New feature
- [x] Code refactor

## Related issue

Closes #1066 

## Screenshot

```
Model: "lstm_autoencoder"
_________________________________________________________________
 Layer (type)            Output Shape         Param #
=================================================================
 encoder (LSTM)          (None, 32)           5248
 encoder_dropout         (None, 32)           0
 repeat_vector           (None, 10, 32)       0
 decoder (LSTM)          (None, 10, 32)       8320
 decoder_dropout         (None, 10, 32)       0
 output (TimeDistributed)(None, 10, 8)        264
=================================================================
```

## Checklist

- [x] I have added screenshots or a recording when they help explain this PR, or noted **N/A** with a one-line reason.
- [x] I have filled in the related issue number when there is one.
- [x] I have tested my changes.
- [x] I have done a self-review of the code.

### CODE REVIEW

**Comment by rudra3007-pro** [comment]
@ronisarkarexe Please review this PR
